.github/workflows/ci.yml

```diff
@@ -0,0 +1,147 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
+      - name: Set up Python
+        run: uv python install 3.11
```

Comment:
Wherever we specify python 3.11 in this file, i'd like a TODO comment pointing to the following issue until we resolve it and move to Python 3.12 or beyond

https://github.com/mitre-attack/mitreattack-python/issues/176